### PR TITLE
Support Swift Macro

### PR DESCRIPTION
**Pull request by qyang-nj**
Make `apple_framework` support swift compiler plugins.

### CODE REVIEW

**Review by luispadron — APPROVED** [review]
Thank you! Mind adding a test fixture that exercises this attr 🙏🏼

**Review by jerrymarino — APPROVED** [review]
🚀

**Review by thiagohmcruz — APPROVED** [review]

File: rules/library.bzl [hunk 1]
```diff
@@ -985,6 +986,7 @@ def apple_library(
             tags = tags_manual,
             defines = defines + swift_defines,
             testonly = testonly,
+            plugins = plugins,
```
Comment (jerrymarino) [inline rules/library.bzl:989]: Perhaps adding this onto `kawrgs` `if plugins != []` will fix the CI issue

File: rules/library.bzl [hunk 2]
```diff
@@ -960,6 +961,9 @@ def apple_library(
     )
 
     if swift_sources:
+        # To be backward compatible with rules_apple 2.x
+        swift_kwargs = dicts.add(kwargs, {"plugins" : plugins} if plugins else {})
```
Comment (qyang-nj) [inline rules/library.bzl:965]: I don't want to modify `kwargs` in place as it may be used in the places later.

**Comment by qyang-nj** [comment]
> Mind adding a test fixture that exercises this attr 🙏🏼
I'm not sure how to add a test just for "rules_apple 3.x" 🤔

**Comment by luispadron** [comment]
> > Mind adding a test fixture that exercises this attr 🙏🏼
> 
> I'm not sure how to add a test just for "rules_apple 3.x" 🤔

@jerrymarino do you have suggestions

**Comment by jerrymarino** [comment]
Is it possible you'd use a `select` statement for the `plugins` attribute you're setting; where the `select` contains a default condition for your plugins but nothing for the LTS stuff?

**Comment by qyang-nj** [comment]
I'm seeing this error. Somehow I cannot use Xcode 15 for this repo. 🤔
```
Error in fail: Swift compiler plugins require Swift 5.9+
```

**Comment by luispadron** [comment]
Ah we need to update CI to support that version. I think this PR is good as-is and we can follow up with a test when we can support it.

**Comment by qyang-nj** [comment]
Sounds good. Thank you. I'm happy to add the tests after the CI supports it.

**Pull request merged**